text/0000-move-analyze-phase.md

```diff
@@ -0,0 +1,111 @@
+# Meta
+[meta]: #meta
+- Name: Move analyze phase
+- Start Date: 2021-01-13
+- Author(s): [@jkutner](github.com/jkutner/) [@jabrown85](github.com/jabrown85)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: [buildpacks/spec#172](https://github.com/buildpacks/spec/pull/172)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+
+This is a proposal to re-order and adjust Lifecycle phases. Specifically, moving "analyze" before "detect".
+
+# Definitions
+[definitions]: #definitions
+
+* __project descriptor__ - the [`project.toml`](https://github.com/buildpacks/spec/blob/main/extensions/project-descriptor.md) extension specification
+
+# Motivation
+[motivation]: #motivation
+
+Doing this would support the following features and capabilities:
+* [Stack buildpacks](https://github.com/buildpacks/rfcs/pull/111), which require a phase to read run-image mixins validation prior to detection
+* [Inline buildpacks](https://github.com/buildpacks/rfcs/blob/main/text/0048-inline-buildpack.md), which require parsing of the `project.toml` in the lifecycle
+* [Lifecycle configuration](https://github.com/buildpacks/rfcs/pull/128)
+
+# What it is
+[what-it-is]: #what-it-is
+
+The analyze phase will now run before the detect phase. The analyze phase will have access to secrets and credentials used to access registries and other services, as it does today. Analyze will no longer require a [`group.toml`](https://github.com/buildpacks/spec/blob/main/platform.md#grouptoml-toml). It will do this by splitting off some of the responsibilites into the restore phase.
+
+## Responsibilities
+
+* Stack validation, to ensure that a new run-image is campatible with the previous app image
+* Retrieve identifier (imageID or digest), stack ID, and mixins, which will be used by subsequent phases
+* Validation of registry credentials, to avoid a long build that fails during export phase
+* Parsing the project descriptor and performance various operations based on its contents, include:
+    - downloading buildpacks
+    - creating ephemeral buildpacks
+    - applying include and exclude rules
+    - adding environment variables to <platform>/env
+    - producing an [`order.toml`](https://github.com/buildpacks/spec/blob/main/platform.md#ordertoml-toml) to be consumed by later phases
+
+## Inputs
+
+* Log level
+* Run-image
+* Stack ID
+* Project descriptor (optional)
```

Comment:
```suggestion
* Project descriptor
```
This isn't the only optional input. We should probably be explicit about which other inputs are optional (most of them), or just remove this to avoid confusing people.